static/app/utils/discover/fieldRenderers.tsx

```diff
@@ -517,6 +531,83 @@ const spanOperationBreakdownRenderer = (field: string) => (
   );
 };
 
+const spanOperationRelativeBreakdownRenderer = (
+  data: EventData,
+  {location}: RenderFunctionBaggage
+): React.ReactNode => {
+  if (
+    !isDurationValue(data, 'spans.total.time') ||
+    SPAN_OP_BREAKDOWN_FIELDS.every(field => !isDurationValue(data, field))
+  ) {
+    return FIELD_FORMATTERS.duration.renderFunc(SPAN_OP_RELATIVE_BREAKDOWN_FIELD, data);
+  }
+
+  const cumulativeSpanOpBreakdown = data['spans.total.time'];
+
+  let otherPercentage = 1;
+
+  return (
+    <RelativeOpsBreakdown>
+      {SPAN_OP_BREAKDOWN_FIELDS.map(field => {
+        const operationName = getSpanOperationName(field) ?? 'op';
+        const spanOpDuration = data[field];
+        const widthPercentage = spanOpDuration / cumulativeSpanOpBreakdown;
```

Comment:
Should we guard against `NaN` here or is it safe

Reply:
@k-fish it's safe since `data` is coming from an API endpoint and `NaN`s cannot cross the network boundary from `json.parse()`.

Reply:
Can there be nulls?

Reply:
`cumulativeSpanOpBreakdown = 0`?;

Reply:
> Can there be nulls?

@Zylphrex should be addressed in https://github.com/getsentry/sentry/pull/25641/commits/f148e58f00417fc24da7536be4d0d1c58324eabd



> `cumulativeSpanOpBreakdown = 0`?;

@k-fish should be addressed in https://github.com/getsentry/sentry/pull/25641/commits/26c9a9a369ba4ce2106d20a5951f67cd195e0896